### PR TITLE
Single Z raise value for all tool-changing / parking

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Remove separate z raise values and unify under a single tool change value.

Add menu entry to adjust

Modify M217 to set for all types of tool changes

Still need to fix a little white space left in the default configs where lines were removed.

If you like this direction, ill get the park positions next.

### CODE REVIEW

**Comment by thinkyhead** [comment]
Since you lack historical perspective on why these various raise values exist, and since there may be valid reasons why these settings were created, I'm not sure we want to accept this. It will take a while to research all the original issues that led to these features being created, so let's just put this on hold.

File: Marlin/src/module/tool_change.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@
     uint8_t singlenozzle_fan_speed[EXTRUDERS];
   #endif
 #endif
-
+float z_raise = TOOLCHANGE_ZRAISE;
```
Comment (thinkyhead) [inline Marlin/src/module/tool_change.cpp]: A global named `z_raise` is a really bad idea.

File: Marlin/src/lcd/ultralcd.cpp [hunk 2]
```diff
@@ -976,13 +976,16 @@ void lcd_quick_feedback(const bool clear_buttons) {
 
   #endif // POWER_LOSS_RECOVERY
 
-  #if ENABLED(SINGLENOZZLE)
```
Comment (thinkyhead) [inline Marlin/src/lcd/ultralcd.cpp]: Making `M217` always enabled even on machines that have only a single extruder makes no sense.

File: Marlin/src/gcode/config/M217.cpp [hunk 3]
```diff
@@ -22,8 +22,6 @@
 
 #include "../../inc/MarlinConfigPre.h"
 
-#if ENABLED(SINGLENOZZLE)
```
Comment (thinkyhead) [inline Marlin/src/gcode/config/M217.cpp:25]: Making `M217` always enabled even on machines that have only a single extruder makes no sense.

File: Marlin/src/gcode/config/M217.cpp [hunk 4]
```diff
@@ -60,20 +60,18 @@ void M217_report(const bool eeprom=false) {
...
-    if (parser.seenval('X')) { report = false; sn_settings.change_point.x = parser.value_linear_units(); }
-    if (parser.seenval('Y')) { report = false; sn_settings.change_point.y = parser.value_linear_units(); }
+    #if ENABLED(SINGLENOZZLE_SWAP_PARK)
+      if (parser.seenval('X')) { report = false; sn_settings.change_point.x = parser.value_linear_units(); }
+      if (parser.seenval('Y')) { report = false; sn_settings.change_point.y = parser.value_linear_units(); }
+    #endif
   #endif
-
-  if (parser.seenval('Z')) { report = false; sn_settings.z_raise = parser.value_linear_units(); }
+  if (parser.seenval('Z')) { report = false; z_raise = parser.value_linear_units(); }
```
Comment (thinkyhead) [inline Marlin/src/gcode/config/M217.cpp]: Making the global `z_raise` always exist even on machines that have only a single extruder makes no sense.

File: Marlin/src/lcd/language/language_en.h [hunk 5]
```diff
@@ -777,8 +777,11 @@
 #ifndef MSG_FILAMENT_SWAP_LENGTH
   #define MSG_FILAMENT_SWAP_LENGTH            _UxGT("Retract Distance")
 #endif
-#ifndef MSG_SINGLENOZZLE_TOOL_CHANGE
-  #define MSG_SINGLENOZZLE_TOOL_CHANGE        _UxGT("Tool Change")
+#ifndef MSG_TOOL_CHANGE
+  #define MSG_TOOL_CHANGE                      _UxGT("Tool Change")
```
Comment (thinkyhead) [inline Marlin/src/lcd/language/language_en.h:781]: Please pay attention to indentation.

File: Marlin/src/module/configuration_store.cpp [hunk 6]
```diff
@@ -1835,11 +1833,12 @@ void MarlinSettings::reset(PORTARG_SOLO) {
     #endif
   #endif
 
+
+  z_raise = TOOLCHANGE_ZRAISE;
```
Comment (thinkyhead) [inline Marlin/src/module/configuration_store.cpp]: Please don't add random blank lines. You should see the amount of cleanup I already did on this PR.

File: Marlin/src/module/tool_change.h [hunk 7]
```diff
@@ -64,6 +63,8 @@
   #endif
 #endif
 
+
```
Comment (thinkyhead) [inline Marlin/src/module/tool_change.h]: Watch don't add random blank lines.

File: Marlin/src/module/tool_change.h [hunk 8]
```diff
@@ -64,6 +63,8 @@
...
+extern float z_raise;
 /**
```
Comment (thinkyhead) [inline Marlin/src/module/tool_change.h:75]: Please pay attention to where blank lines are added. This should have no extra blank line in front, but should have a new blank line after. You're not doing these things just to troll me, are you? 🙄

**Comment by InsanityAutomation** [comment]
This doesn't eliminate any, just combines them under a single setting instead of a bunch of different ones. Instead of z_raise being independent, my last thought was to make a generic took change struct and combine park for each type that uses it as well. No trolling on spaces, just a concept shot to talk about before I finish up later. My find replace of the settings adding spaces annoyed me too!

**Comment by thinkyhead** [comment]
I'll look it over more closely. My pneumonia, car being in the shop for a week, looming medical bills, new dumb housemates, and annoying brainless clients have all got me feeling cranky today.

How are you??

**Pull request merged**